srcpkgs/mold/patches/00-makefile.patch

```diff
@@ -0,0 +1,84 @@
+diff --git a/Makefile b/Makefile
+index 207f54c..e20a55d 100644
+--- a/Makefile
++++ b/Makefile
+@@ -1,15 +1,14 @@
+ CC = clang
+ CXX = clang++
+ 
+-MIMALLOC_LIB = mimalloc/out/release/libmimalloc.a
+-GIT_HASH = $(shell [ -d .git ] && git rev-parse HEAD)
++GIT_HASH ?= $(shell [ -d .git ] && git rev-parse HEAD)
+ 
+ CPPFLAGS = -g -Imimalloc/include -pthread -std=c++20 \
+            -Wno-deprecated-volatile \
+            -DMOLD_VERSION=\"0.9.1\" \
+            -DGIT_HASH=\"$(GIT_HASH)\" \
+ 	   $(EXTRA_CPPFLAGS)
+-LDFLAGS = $(EXTRA_LDFLAGS)
++LDFLAGS += $(EXTRA_LDFLAGS)
+ LIBS = -Wl,-as-needed -lcrypto -pthread -ltbb -lz -lxxhash -ldl
+ OBJS = main.o object_file.o input_sections.o output_chunks.o \
+        mapfile.o perf.o linker_script.o archive_file.o output_file.o \
+@@ -41,7 +40,12 @@ else
+   # By default, we want to use mimalloc as a memory allocator.
+   # Since replacing the standard malloc is not compatible with ASAN,
+   # we do that only when ASAN is not enabled.
+-  LDFLAGS += -Wl,-whole-archive $(MIMALLOC_LIB) -Wl,-no-whole-archive
++  ifdef SYSTEM_MIMALLOC
++    LIBS += -lmimalloc
++  else
++    MIMALLOC_LIB = mimalloc/out/release/libmimalloc.a
++    LIBS += -Wl,-whole-archive $(MIMALLOC_LIB) -Wl,-no-whole-archive
++  endif
+ endif
+ 
+ ifeq ($(TSAN), 1)
+@@ -52,7 +56,7 @@ endif
+ all: mold mold-wrapper.so
+ 
+ mold: $(OBJS) $(MIMALLOC_LIB)
+-	$(CXX) $(CFLAGS) $(OBJS) -o $@ $(LDFLAGS) $(LIBS)
++	$(CXX) $(CXXFLAGS) $(OBJS) -o $@ $(LDFLAGS) $(LIBS)
+ 
+ mold-wrapper.so: mold-wrapper.c Makefile
+ 	$(CC) -fPIC -shared -o $@ $< -ldl
+@@ -62,26 +66,27 @@ $(OBJS): mold.h elf.h Makefile
+ $(MIMALLOC_LIB):
+ 	mkdir -p mimalloc/out/release
+ 	(cd mimalloc/out/release; CFLAGS=-DMI_USE_ENVIRON=0 cmake ../..)
+-	$(MAKE) -C mimalloc/out/release
++	$(MAKE) -C mimalloc/out/release mimalloc-static
+ 
+ test tests check: all
+ 	 $(MAKE) -C test --output-sync --no-print-directory
+ 
+ install: all
+-	install -m 755 mold $(PREFIX)/bin
+-	strip $(PREFIX)/bin/mold
++	install -m 755 mold $(DESTDIR)$(PREFIX)/bin
++	strip $(DESTDIR)$(PREFIX)/bin/mold
+ 
+-	install -m 755 -d $(PREFIX)/lib/mold
+-	install -m 644 mold-wrapper.so $(PREFIX)/lib/mold
+-	strip $(PREFIX)/lib/mold/mold-wrapper.so
++	install -m 755 -d $(DESTDIR)$(PREFIX)/lib/mold
++	install -m 644 mold-wrapper.so $(DESTDIR)$(PREFIX)/lib/mold
++	strip $(DESTDIR)$(PREFIX)/lib/mold/mold-wrapper.so
```

Comment:
You can run make install with `PREFIX=$DESTDIR$PREFIX` and avoid the patch. Though it should be upstreamed at some point.

Reply:
It's upstreamed at this moment, but no new tag released yet. I can ask him for a tag though.

Reply:
Waiting for rui314/mold#65